src/lib.rs

```diff
@@ -11,6 +11,678 @@
+
+        // Map UAttributes.id
+        let id_val = attributes.id.as_ref().ok_or(UStatus::fail_with_code(
+            UCode::INTERNAL,
+            "No UAtrributes id found",
+        ))?;
+        properties
+            .push_string_pair(
+                mqtt::PropertyCode::UserProperty,
+                ID_NUM,
+                &id_val.to_hyphenated_string(),
+            )
+            .map_err(|_| UStatus::fail_with_code(UCode::INTERNAL, "Unable to add id property"))?;
+
+        // Map UAttributes.type
+        let type_val = attributes.type_.value();
+        properties
+            .push_string_pair(
+                mqtt::PropertyCode::UserProperty,
+                TYPE_NUM,
+                &type_val.to_string(),
+            )
+            .map_err(|_| UStatus::fail_with_code(UCode::INTERNAL, "Unable to add type property"))?;
+
+        // Map UAttributes.source
+        let source_val = attributes.source.as_ref().ok_or(UStatus::fail_with_code(
+            UCode::INTERNAL,
+            "No UAtrributes source found",
+        ))?;
+        let source_string: String = source_val.into();
+        properties
+            .push_string_pair(mqtt::PropertyCode::UserProperty, SOURCE_NUM, &source_string)
+            .map_err(|_| {
+                UStatus::fail_with_code(UCode::INTERNAL, "Unable to add source property")
+            })?;
+
+        // Map UAttributes.sink
+        let msg_type = up_rust::UMessageType::from_i32(type_val).unwrap();
+        if msg_type != up_rust::UMessageType::UMESSAGE_TYPE_PUBLISH {
+            let sink_val = attributes.sink.as_ref().ok_or(UStatus::fail_with_code(
+                UCode::INTERNAL,
+                "No UAtrributes sink found",
+            ))?;
+            let sink_string: String = sink_val.into();
+            properties
+                .push_string_pair(mqtt::PropertyCode::UserProperty, SINK_NUM, &sink_string)
+                .map_err(|_| {
+                    UStatus::fail_with_code(UCode::INTERNAL, "Unable to add sink property")
+                })?;
+        }
+
+        // Map UAttributes.priority
+        let priority_enum = attributes
+            .priority
+            .enum_value()
+            .map_err(|_| UStatus::fail_with_code(UCode::INTERNAL, "Unable to parse priority"))?;
+
+        // If priority code is unspecified, default to CS1.
+        let priority_val = if priority_enum == UPriority::UPRIORITY_UNSPECIFIED {
+            UPriority::UPRIORITY_CS1.to_priority_code()
+        } else {
+            priority_enum.to_priority_code()
+        };
+
+        properties
+            .push_string_pair(
+                mqtt::PropertyCode::UserProperty,
+                PRIORITY_NUM,
+                &priority_val,
+            )
+            .map_err(|_| {
+                UStatus::fail_with_code(UCode::INTERNAL, "Unable to add priority property")
+            })?;
+
+        // Map Optional UAttributes.ttl
+        if let Some(ttl_val) = attributes.ttl {
```

Comment:
IIRC TTL is required for Request mesages

Reply:
Per one of your suggestions, added the use of the Uattributes validator from up-rust to ensure uattributes that are processed adhere to the uMessage conventions